### ISSUE TITLE
[MCHANGES-409] allow to completely skip github report

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-changes-plugin#279]
**[Luc Maisonobe](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=luc)** opened **[MCHANGES-409](https://issues.apache.org/jira/browse/MCHANGES-409?redirect=false)** and commented

When using changes plugin on a project hosted at GitHub, there is always

an attempt to connect to GitHub even when reportSet contains **only** changes-report and nothing related to GitHub.

 

This creates several problems. The first one was already identified with MCHANGES-378 with

the subsequents request MCHANGES-396 which are not yet solved.

 

A new one arose in recent days as GitHub does not support username/password authentication anymore and MCHANGES-400 seems to prevent connecting otherwise.

 

So basically for now, being hosted at GitHub completely prevents using maven-changes-plugin since it mandates connecting to an host that mandates an authentication method that is not supported. And this even when you do **not** want the GitHub report at all.

 

So it would be nice to have a configuration parameter that allows to completely skip attempts to connect to GitHub (and the same would probably be useful also for other reports)

---

**Affects:** 2.12.1

**Comment by jira-importer** [comment]
**[Slawomir Jaranowski](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=slawekjaranowski)** commented

When reportSet contains **only** changes-report, githhub-report and other don't executed.

Can you provide a simple configuration which reproduce it?

**Comment by jira-importer** [comment]
**[Slawomir Jaranowski](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=slawekjaranowski)** commented

So I close issue - if you will reproduce let me know
